### PR TITLE
Dept 4 search api #6

### PR DESCRIPTION
**Pull request by nkhorvash**
This is a port of the team1 search proxy API

I've also added the `.prettierrc` file to the project root to configure prettier for the project

### CODE REVIEW

File: pages/api/profile/search.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    const response = await fetch(url, {
+      headers: {
+        authorization: `Bearer ${accessToken}`,
+      },
+    });
+
+    if (response.status === 200) {
+      let profiles = (await response.json()) as Array<BasicProfile>;
+
+      if (profiles.length > 100) {
```
Comment (bdurel) [inline pages/api/profile/search.ts:40]: Do we need this here if the backend is already putting a cap at 100?

**Review by bdurel — CHANGES_REQUESTED** [review]
Just a couple comments to address.

File: pages/api/profile/search.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { getAccessToken, withApiAuthRequired } from '@auth0/nextjs-auth0';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { BasicProfile } from 'ts/types';
+
+/**
+ * @name profileBySearch
+ * @description makes a request to an external API for the basic profiles that match a given name/email
+ * @returns 200 with up to 100 profiles if successful, other status codes if given other information
+ */
+export default withApiAuthRequired(async function profileBySearch(
```
[the remark this answers is not in the document]
Comment (nkhorvash) [reply pages/api/profile/search.ts:11]: Or what about profileSearch, it's still singular, but maybe addresses your point. Or maybe it doesn't, I hate naming, lol. Let me know, I'll put whatever you suggest.

[the remark this answers is not in the document]
Comment (bdurel) [reply pages/api/profile/search.ts:11]: We currently have these two API functions: latestShoutouts and userProfile. Seems like the naming convention is to name the function by what is returned. So profilesBySearch is most readable to me, but I would be fine with profilesSearch or even searchProfiles.

[the remark this answers is not in the document]
Comment (nkhorvash) [reply pages/api/profile/search.ts:11]: good point, sometimes I go in circles in my head about what is more readable, consistency is an important piece of readability

**Review by bdurel — APPROVED** [review]

**Review by agonxalez — APPROVED** [review]
Looks good to me, nice job.

**Pull request merged**